lib/msal-browser/src/interaction_client/WamInteractionClient.ts

```diff
@@ -0,0 +1,174 @@
+import { EventHandler } from "../event/EventHandler";
+import { EndSessionRequest } from "../request/EndSessionRequest";
+import { PopupRequest } from "../request/PopupRequest";
+import { SilentRequest } from "../request/SilentRequest";
+import { SsoSilentRequest } from "../request/SsoSilentRequest";
+import { WamMessageHandler } from "../broker/wam/WamMessageHandler";
+import { WamExtensionMethod } from "../utils/BrowserConstants";
+import { WamExtensionRequestBody, WamTokenRequest } from "../broker/wam/WamRequest";
+import { WamResponse } from "../broker/wam/WamResponse";
+import { WamAuthError } from "../error/WamAuthError";
+
+export class WamInteractionClient extends BaseInteractionClient {
+    protected provider: WamMessageHandler;
+
+    constructor(config: BrowserConfiguration, browserStorage: BrowserCacheManager, browserCrypto: ICrypto, logger: Logger, eventHandler: EventHandler, provider: WamMessageHandler, correlationId?: string) {
+        super(config, browserStorage, browserCrypto, logger, eventHandler, correlationId);
+        this.provider = provider;
+    }
+
+    /**
+     * Acquire token from WAM via browser extension
+     * @param request 
+     */
+    async acquireToken(request: PopupRequest|SilentRequest|SsoSilentRequest): Promise<AuthenticationResult> {
+        this.logger.trace("WamInteractionClient - acquireToken called.");
+        const wamRequest = this.initializeWamRequest(request);
+
+        const messageBody: WamExtensionRequestBody = {
+            method: WamExtensionMethod.GetToken,
+            request: wamRequest
+        };
+
+        const reqTimestamp = TimeUtils.nowSeconds();
+        const response: object = await this.provider.sendMessage(messageBody);
+        this.validateWamResponse(response);
+        return this.handleWamResponse(response as WamResponse, wamRequest, reqTimestamp);
+    }
+
+    /**
+     * Logout from WAM via browser extension
+     * @param request 
+     */
+    logout(request?: EndSessionRequest): Promise<void> {
+        this.logger.trace("WamInteractionClient - logout called.");
+        return Promise.reject("Logout not implemented yet");
+    }
+
+    /**
+     * Transform response from WAM into AuthenticationResult object which will be returned to the end user
+     * @param response 
+     * @param request 
+     * @param reqTimestamp 
+     */
+    protected async handleWamResponse(response: WamResponse, request: WamTokenRequest, reqTimestamp: number): Promise<AuthenticationResult> {
+        this.logger.trace("WamInteractionClient - handleWamResponse called.");
+        // create an idToken object (not entity)
+        const idTokenObj = new AuthToken(response.id_token || Constants.EMPTY_STRING, this.browserCrypto);
+
+        // Save account in browser storage
+        const homeAccountIdentifier = AccountEntity.generateHomeAccountId(response.client_info || Constants.EMPTY_STRING, AuthorityType.Default, this.logger, this.browserCrypto, idTokenObj);
```

Comment:
AuthorityType.Default does not cover ADFS. Are we not supporting ADFS for WAM?

Reply:
No, can't support direct connection to ADFS. Federated ADFS, however, would be covered by AuthorityType.Default as it goes through AAD